Add migration guide pointers for removed policies, tokenizers, and featurizers. (#10171)
RasaHQ/rasa: docs/docs/migration-guide.mdx

However, the previous behavior can still be enforced by setting
`ranking_length=10` and `renormalize_confidences=True`.

#### Removed Policies

Several dialogue policies that were deprecated in Rasa 2.x have been removed in Rasa 3.x.
If you are trying to migrate a config file with a removed policy,
consult the migration guides for the individual policies:

- `FallbackPolicy` [migration guide](#manually-migrating-from-the-fallback-policy)
- `TwoStageFallbackPolicy` [migration guide](#manually-migrating-from-the-two-stage-fallback-policy)
- `MappingPolicy` [migration guide](#manually-migrating-from-the-mapping-policy)
- `FormPolicy` [migration guide](#forms)

> - `SklearnPolicy` should be replaced with [TedPolicy](/policies#ted-policy)

Reviewer: ```suggestion
- `SklearnPolicy` should be replaced with [TEDPolicy](/policies#ted-policy)
```

However, the previous behavior can still be enforced by setting
`ranking_length=10` and `renormalize_confidences=True`.

#### Removed Policies

Several dialogue policies that were deprecated in Rasa 2.x have been removed in Rasa 3.x.

> If you are trying to migrate a config file with a removed policy,

Reviewer: ```suggestion
If you are migrating a config file with a removed policy,
```

> consult the migration guides for the individual policies:

Reviewer: ```suggestion
consult the following migration guides for the individual policies:
```

- `FallbackPolicy` [migration guide](#manually-migrating-from-the-fallback-policy)
- `TwoStageFallbackPolicy` [migration guide](#manually-migrating-from-the-two-stage-fallback-policy)
- `MappingPolicy` [migration guide](#manually-migrating-from-the-mapping-policy)
- `FormPolicy` [migration guide](#forms)

> - `SklearnPolicy` should be replaced with [TEDPolicy](/policies#ted-policy)

Reviewer: Is there any indication of which settings to use when migrating? Even an entry to the changelog would help since there is no other info on this

Author: You are referring to the SklearnPolicy specifically or all of the above policies?

Reviewer: sklearn specifically, my bad

Author: @melindaloubser1 I can't find anything on SklearnPolicy other than [this changelog](https://rasa.com/docs/rasa/changelog#deprecations-and-removals-6) entry (see #6658). The issue linked in the change log entry doesn't say anything about why or how it should be replaced.

I think the SklearnPolicy can take arbitrary sklearn models as an argument so it will be hard to give very specific guidance. 

How does the following strike you:

> - `SklearnPolicy` should be replaced with [TEDPolicy](/policies#ted-policy). We recommend the [default TED config](https://rasa.com/docs/rasa/model-configuration#suggested-config) as a starting point.

Reviewer: Gotcha, in that case yes sounds good

Author: Sweet!

#### Removed Tokenizers and Featurizers

The `ConveRTTokenizer`, `LanguageModelTokenizer`, and `HFTransformersNLP` featurizer

> components were deprecated in Rasa 2.x and have been removed in Rasa 3.x. See this
> [migration guide](#deprecations-2) for replacing these components in your pipeline.  

Reviewer: ```suggestion
components were deprecated in Rasa 2.x and have been removed in Rasa 3.x. See the
[migration guide for Rasa 2.x](#deprecations-2) for replacing these components in your pipeline.  
```
